language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FunctionCompletionItem.java

```diff
@@ -0,0 +1,40 @@
+/*
+ *  Copyright (c) 2023, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package org.ballerinalang.langserver.completions.providers.context;
+
+import org.ballerinalang.langserver.commons.BallerinaCompletionContext;
+import org.ballerinalang.langserver.commons.completion.AbstractLSCompletionItem;
+import org.eclipse.lsp4j.CompletionItem;
+
+/**
+ * Completion item for main function.
+ *
+ * @since 2201.4.0
+ */
+public class FunctionCompletionItem extends AbstractLSCompletionItem {
```

Comment:
Having a `FunctionCompletionItem` can be confusing as we don't use this to only main function completion item. Is there a specific reason to use a separate class for this without using the FunctionCompletionItemBuilder other than the sorting purposes? If not shall we explore the possibility of using the `SnippetCompletionItem`

Reply:
As I remember the reason was that we cannot match the constructor of `SnippetCompletionItem` here because we don't have a snippet block. `main()` function snippet is not static anymore due to handling the presence of `public` keyword.

Reply:
It was originally named MainFunctionCompletionItem and @IMS94 suggested that it would be better to rename it to FunctionCompletionItem as it doesn't contain anything specific to the main function. WDYT?

Reply:
We can address this later as this is not a critical improvement